### PR TITLE
Handle spaces in repo URIs

### LINKED ISSUE #15618: References panel fetches highlighted file without %-decoding repo from URI
Repro: https://k8s.sgdev.org/sourcegraph.visualstudio.com/Test%20Repo/-/blob/Test%20Folder/Test%20File.cs#L7:20&tab=references

image

The highlight request includes the `%` from the URI of the repo (interestingly not of the file):

image

We should add an extension integration test for this.

### PR DESCRIPTION
**Pull request by felixfbecker**
Fixes #15618

### CODE REVIEW

File: client/shared/src/util/url.ts [hunk 1]
```diff
@@ -517,24 +517,28 @@ export function toViewStateHashComponent(viewState: string | undefined): string
 const positionString = (position: Position): string =>
     position.line.toString() + (position.character ? `,${position.character}` : '')
 
+export const encodeURIComponentExceptSlashes = (component: string): string =>
```
Comment (tjkandala) [inline client/shared/src/util/url.ts:520]: Should we use this in `RepoContainer` to construct `repoMatchURL`?

**Review by tjkandala — APPROVED** [review]

File: client/shared/src/util/url.ts (hunk 1 above)
Comment (felixfbecker) [reply client/shared/src/util/url.ts:520]: Yep, done!

**Comment by felixfbecker** [comment]
@sourcegraph/web I made a few more updates

**Review by tjkandala — APPROVED** [review]
Latest changes look good, but it feels weird to have to encode `repoName` for every link. 

I believe that if we pass an array with the encoded and unencoded `repoName` to `Route`s path prop, all links could work. That may not be the best solution (might not even work, but I remember it worked when I manually tested it), but we're easily at risk of introducing bad links in the future.

**Review by marekweb — APPROVED** [review]

**Comment by felixfbecker** [comment]
@tjkandala I think you made some valid points, it is concerning that we may forget this for every link. Thoughts:
- If we match both, we could have two states in the app for the same situation, one where `props.location` has spaces in it and one where it doesn't. I would be worried about having to think about both possible states.
- It would also mean that `props.location.path` is not always the same as `window.location.path`, which could break some logic.
- We have this problem not just with repo names, but also with file names, so just including it in the repo URL matching wouldn't protect against everything. We'd still have to make sure all link URLs encode properly.
- Maybe the solution could be some kind of lint rule protecting against raw string interpolation to `<Link>`.

**Pull request merged**